framework/src/controller/channels/in_memory_channel.ts

```diff
@@ -120,4 +145,14 @@ export class InMemoryChannel extends BaseChannel {
 		const resp = await this.bus.invoke<T>(request.toJSONRPCRequest());
 		return resp.result;
 	}
+
+	private _getModuleDB(): Database {
+		if (this._moduleDB) {
+			return this._moduleDB;
+		}
+
+		const { data: dbFolder } = systemDirs(this.config.label, this.config.rootPath);
+		this._moduleDB = new Database(path.join(dbFolder, 'module.db'));
```

Comment:
the database cannot be opened multiple times. the DB instance needs to be passed down to this class through constructor similar to the `_db`